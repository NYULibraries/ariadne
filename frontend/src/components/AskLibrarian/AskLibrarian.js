import PropTypes from 'prop-types';

const AskLibrarian = () => {
    return (
        <div className="ask-librarian">
            <h2>Need Help?</h2>
            <h3>
                <a href={ASK_LIBRARIAN_URL} target="_blank" rel="noopener noreferrer">
                    {ASK_LIBRARIAN_TEXT}
                </a>
            </h3>
            <div>
                <p>
                    Use <a href="https://library.nyu.edu/ask/" target="_blank" rel="noreferrer">Ask A Librarian</a> or the &quot;Chat with Us&quot; icon at the bottom right corner for any question you have about the Libraries&apos; services.
                </p>
                <p>
                    Visit our <a href="https://guides.nyu.edu/online-tutorials/finding-sources" target="_blank" rel="noreferrer">online tutorials</a> for tips on searching the catalog and getting library resources.
                </p>
                <h3>Additional Resources</h3>
                <ul>
                    <li>Use <a href="https://ezborrow.reshare.indexdata.com/" target="_blank" rel="noreferrer">EZBorrow</a> or <a href="https://library.nyu.edu/services/borrowing/from-non-nyu-libraries/interlibrary-loan/" target="_blank" rel="noreferrer">InterLibrary Loan (ILL)</a> for materials unavailable at NYU</li>
                    <li>Discover subject specific resources using <a href="http://guides.nyu.edu" target="_blank" rel="noreferrer">expert curated research guides</a></li>
                    <li>Explore the <a href="https://library.nyu.edu/services/" target="_blank" rel="noreferrer">complete list of library services</a></li>
                    <li>Reach out to the Libraries on <a href="https://www.instagram.com/nyulibraries/" target="_blank" rel="noreferrer">our Instagram</a></li>
                    <li>Search <a href="https://www.worldcat.org/search?qt=worldcat_org_all" target="_blank" rel="noreferrer">WorldCat</a> for items in nearby libraries</li>
                </ul>
            </div>
        </div>
    );
};


AskLibrarian.propTypes = {
    loading: PropTypes.bool.isRequired
};

export const ASK_LIBRARIAN_TEXT = 'Ask a Librarian';
export const ASK_LIBRARIAN_URL = 'https://library.nyu.edu/ask/';

export default AskLibrarian;
