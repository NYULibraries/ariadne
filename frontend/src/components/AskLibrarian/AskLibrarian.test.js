import AskLibrarian, { ASK_LIBRARIAN_TEXT, ASK_LIBRARIAN_URL } from './AskLibrarian';

import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

describe('AskLibrarian component', () => {
    it('renders "Need Help?" when not loading', () => {
        const { getByText } = render(<AskLibrarian />);
        expect(getByText('Need Help?')).toBeInTheDocument();
    });

    it('renders the correct ASK_LIBRARIAN_TEXT and ASK_LIBRARIAN_URL', () => {
        const { getByText } = render(<AskLibrarian />);
        const link = getByText(ASK_LIBRARIAN_TEXT);
        expect(link).toBeInTheDocument();
        expect(link.getAttribute('href')).toBe(ASK_LIBRARIAN_URL);
    });

    it('renders additional resources when not loading', () => {
        const { queryAllByText } = render(<AskLibrarian />);
        expect(queryAllByText(/Additional Resources/).length).toBeGreaterThan(0);
    });


    it('opens the ASK_A_LIBRARIAN_URL in a new tab when the link is clicked', () => {
        const { getByText } = render(<AskLibrarian />);
        const link = getByText(ASK_LIBRARIAN_TEXT);
        userEvent.click(link);
        expect(link).toHaveAttribute('target', '_blank');
        expect(link).toHaveAttribute('rel', 'noopener noreferrer');
        expect(link).toHaveAttribute('href', ASK_LIBRARIAN_URL);
    });

    it('matches snapshot', () => {
        const { container } = render(<AskLibrarian />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
