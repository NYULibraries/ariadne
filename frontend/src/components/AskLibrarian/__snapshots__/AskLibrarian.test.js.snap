// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskLibrarian component matches snapshot 1`] = `
<div
  class="ask-librarian"
>
  <h2>
    Need Help?
  </h2>
  <h3>
    Ask a Librarian
  </h3>
  <div>
    <p>
      Use 
      <a
        href="https://library.nyu.edu/ask/"
        rel="noreferrer"
        target="_blank"
      >
        Ask a Librarian
      </a>
       or the "Chat with Us" icon at the bottom right corner for any question you have about the Libraries' services.
    </p>
    <p>
      Visit our 
      <a
        href="https://guides.nyu.edu/online-tutorials/finding-sources"
        rel="noreferrer"
        target="_blank"
      >
        online tutorials
      </a>
       for tips on searching the catalog and getting library resources.
    </p>
    <h3>
      Additional Resources
    </h3>
    <ul>
      <li>
        Use 
        <a
          href="https://ezborrow.reshare.indexdata.com/"
          rel="noreferrer"
          target="_blank"
        >
          EZBorrow
        </a>
         or 
        <a
          href="https://library.nyu.edu/services/borrowing/from-non-nyu-libraries/interlibrary-loan/"
          rel="noreferrer"
          target="_blank"
        >
          InterLibrary Loan (ILL)
        </a>
         for materials unavailable at NYU
      </li>
      <li>
        Discover subject specific resources using 
        <a
          href="http://guides.nyu.edu"
          rel="noreferrer"
          target="_blank"
        >
          expert curated research guides
        </a>
      </li>
      <li>
        Explore the 
        <a
          href="https://library.nyu.edu/services/"
          rel="noreferrer"
          target="_blank"
        >
          complete list of library services
        </a>
      </li>
      <li>
        Search 
        <a
          href="https://www.worldcat.org/search?qt=worldcat_org_all"
          rel="noreferrer"
          target="_blank"
        >
          WorldCat
        </a>
         for items in nearby libraries
      </li>
      <li>
        <a
          href="https://nyu.qualtrics.com/jfe/form/SV_a30SvEFe8f1yXQh"
          rel="noreferrer"
          target="_blank"
        >
          Problem with an e-resource? Fill out this form
        </a>
      </li>
      <li>
        <a
          href="https://nyu.qualtrics.com/jfe/form/SV_blQ3OFOew9vl6Pb?Source=NYU"
          rel="noreferrer"
          target="_blank"
        >
          Submit feedback about GetIt
        </a>
      </li>
    </ul>
  </div>
</div>
`;
